NasdaqDataLink.cs

```diff
@@ -0,0 +1,194 @@
+        /// </summary>
+        public static bool IsAuthCodeSet
+        {
+            get;
+            private set;
+        }
+        
+        /// <summary>
+        /// Default NasdaqDataLink constructor uses Close as its value column
+        /// </summary>
+        public NasdaqDataLink() : this("Close")
+        {
+        }
+
+        /// <summary>
+        /// Constructor for creating customized NasdaqDataLink instance which doesn't use "Close" as its value item.
+        /// </summary>
+        /// <param name="valueColumnName"></param>
+        protected NasdaqDataLink(string valueColumnName)
+        {
+            _valueColumn = valueColumnName;
+        }
+
+        /// <summary>
+        /// Using the Nasdaq Data Link V3 API automatically set the URL for the dataset.
+        /// </summary>
+        /// <param name="config">Subscription configuration object</param>
+        /// <param name="date">Date of the data file we're looking for</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>STRING API Url for Nasdaq Data Link.</returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            var source = $"https://data.nasdaq.com/api/v3/datasets/{config.Symbol.Value}.csv?order=asc&api_key={_authCode}";
+            return new SubscriptionDataSource(source, SubscriptionTransportMedium.RemoteFile);
+        }
+
+        /// <summary>
+        /// Parses the data from the line provided and loads it into LEAN
+        /// </summary>
+        /// <param name="config">Subscription configuration</param>
+        /// <param name="line">Line of data</param>
+        /// <param name="date">Date</param>
+        /// <param name="isLiveMode">Is live mode</param>
+        /// <returns>New instance</returns>
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            // be sure to instantiate the correct type
+            var data = (NasdaqDataLink) Activator.CreateInstance(GetType());
+            data.Symbol = config.Symbol;
+            var csv = line.Split(',');
+
+            if (!_isInitialized)
+            {
+                _isInitialized = true;
+                foreach (var propertyName in csv)
+                {
+                    var property = propertyName.Trim().ToLowerInvariant();
+                    data.SetProperty(property, 0m);
+                    _propertyNames.Add(property);
+                }
+                // Returns null at this point where we are only reading the properties names
+                return null;
+            }
+
+            data.Time = DateTime.ParseExact(csv[0], "yyyy-MM-dd", CultureInfo.InvariantCulture);
+            data.EndTime = data.Time + TimeSpan.FromDays(1);
```

Comment:
I think we should use
```csharp
data.EndTime = data.Time + config.Resolution.ToTimeSpan()
```
unless we are 100% sure that all data has daily resolution. Given the [default is Resolution.Daily](https://github.com/QuantConnect/Lean.DataSource.NasdaqDataLink/pull/1/files#diff-a58be19fb84f78ea7c46024420b29dc02a00e57eb039ca0e26cf62976f374564R174), it will likely fall in this case, but we give more options here.